src/applications/gi/containers/SearchPage.jsx

```diff
@@ -161,12 +165,31 @@ export class SearchPage extends React.Component {
 
     const shouldRemoveFilter =
       !value ||
-      ((field === 'country' || field === 'state' || field === 'type') &&
+      ((field === 'country' ||
+        field === 'state' ||
+        field === 'type' ||
+        field === 'relaffil') &&
         value === 'ALL');
 
     if (shouldRemoveFilter) {
       delete query[field];
     }
+
+    if (field === 'gender') {
+      if (value === 'womenonly') {
+        query = { ...query, womenonly: 'true' };
+        delete query?.menonly;
+      }
+      if (value === 'menonly') {
+        query = { ...query, menonly: 'true' };
+        delete query?.womenonly;
+      }
+      if (value === 'Any') {
+        delete query?.menonly;
+        delete query?.womenonly;
+      }
+      delete query?.gender;
+    }
```

Comment:
I think this could be simplified. I would delete menonly and womenonly at the top, then add menonly or womenonly if selected.

Reply:
That is a good suggestion. I can clean this bit of logic up here a little bit.